### PR TITLE
Feat/make i18n handle language storage

### PR DESCRIPTION
**Pull request by lolottetheclash**
The purpose of this PR is to stop using query strings to modify the language and to let i18n manage entirely the language in the local storage.

### CODE REVIEW

File: stories/guidelines/i18n/I18n.stories.mdx [hunk 1]
```diff
@@ -81,38 +106,135 @@ type I18nResource = Readonly<{
 }>
 ```
 
+#### The 3 ways to implement translations explained step by step
+
+The `LanguageSwitcher`component comes with its own translations which are stored in the `LanguageSwitcher_en.json` file.
```
Comment (fredericvilcot) [inline stories/guidelines/i18n/I18n.stories.mdx:111]: As this part of the documentation is fully agnostic about other UI components, we could rather mention a fake component as example (more neutral), what do you think ? 
```suggestion
Given an `AwesomeComponent` that comes with its own translations which are stored in the `AwesomeComponent_en.json` file.  
```

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
That's a great enhancement, thx! 😎
I have noticed some remarks, it's up to you!

File: stories/guidelines/i18n/I18n.stories.mdx [hunk 2]
```diff
@@ -42,15 +32,19 @@ i18nextLng=LANGUAGE
 It is thus quite possible to update this value according to a user interaction and to force a refresh of the translations and i18n keys.
 For example:
 
-```typescript
-import i18n from '@okp4/ui'
+```jsx
```
[the remark this answers is not in the document]
Comment (lolottetheclash) [reply stories/guidelines/i18n/I18n.stories.mdx:35]: That's right!

File: stories/guidelines/i18n/I18n.stories.mdx [hunk 3]
```diff
@@ -42,15 +32,19 @@ i18nextLng=LANGUAGE
...
+import updateLanguage from '@okp4/ui'
```
[the remark this answers is not in the document]
Comment (lolottetheclash) [reply stories/guidelines/i18n/I18n.stories.mdx:36]: Good catch!

File: stories/guidelines/i18n/I18n.stories.mdx [hunk 4]
```diff
@@ -59,19 +53,50 @@ const updateLanguage = (user: { name: string; language: string }): void => {
...
+
+### Check current language
+
+---
+
+The `isCurrentLanguage` method can be used to facilitate the detection of the currently used language and improve user experience.  
+This method takes a language as parameter and returns a boolean if the language detected is the current language.
+
+```ts
+ const isCurrentLanguage = (lng: string): boolean
```
[the remark this answers is not in the document]
Comment (lolottetheclash) [reply stories/guidelines/i18n/I18n.stories.mdx:66]: Very different yes!

File: stories/guidelines/i18n/I18n.stories.mdx [hunk 5]
```diff
@@ -59,19 +53,50 @@ const updateLanguage = (user: { name: string; language: string }): void => {
...
+
+An example of usage in a React component:
+
+##### **`MyComponent.tsx`**
+
+```tsx
+import React from 'react'
+
+export const MyComponent: React.FC = () => {
+  return <p>{isCurrentLanguage('fr') ? 'French Language On' : 'French Language Off'}</p>
```
[the remark this answers is not in the document]
Comment (lolottetheclash) [reply stories/guidelines/i18n/I18n.stories.mdx:77]: Indeed...

File: stories/guidelines/i18n/I18n.stories.mdx (hunk 1 above)
Comment (lolottetheclash) [reply stories/guidelines/i18n/I18n.stories.mdx:111]: Absolutely right, thanks !

**Review by ErikssonJoakim — DISMISSED** [review]
Looking good 💪

**Review by fredericvilcot — APPROVED** [review]
LGTM!

**Review by fredericvilcot — APPROVED** [review]

**Review by ccamel — APPROVED** [review]

**Pull request merged**